### ISSUE TITLE
Issues with socket.sendall (0.6.5/0.7.0)

### ISSUE THREAD

**Issue by dcramer** [gabrielfalcao/HTTPretty#113]
This doesn't seem to happen in every case, but I'm hitting an infinite loop with the sendall functionality (seems to be around line 237 in 0.6.5):

This is simply from flushing a SELECT statement in SQLA (in 0.6.5):

```
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
recvfrom(0xA, 0x10BB6B9BC, 0x10)		 = -1 Err#35
```

On 0.7.0 I'm seeing this, which seems to be related:

```
changes/ext/pubsub.py:44: in publish
>       self._redis.publish(channel, json.dumps(data))
../../.virtualenvs/changes/lib/python2.7/site-packages/redis/client.py:1447: in publish
>       return self.execute_command('PUBLISH', channel, message)
../../.virtualenvs/changes/lib/python2.7/site-packages/redis/client.py:393: in execute_command
>           connection.send_command(*args)
../../.virtualenvs/changes/lib/python2.7/site-packages/redis/connection.py:306: in send_command
>       self.send_packed_command(self.pack_command(*args))
../../.virtualenvs/changes/lib/python2.7/site-packages/redis/connection.py:290: in send_packed_command
>           self._sock.sendall(command)
../../.virtualenvs/changes/lib/python2.7/site-packages/httpretty/core.py:354: in sendall
>               return self.real_sendall(data, *args, **kw)
../../.virtualenvs/changes/lib/python2.7/site-packages/httpretty/core.py:331: in real_sendall
>               received = self.truesock.recv(self._bufsize)
E               ValueError: negative buffersize in recv
```

**Comment by yrik** [comment]
+1 redis and moto (httpretty)

**Comment by yrik** [comment]
@Frozenball could you show your patch?

**Comment by Frozenball** [comment]
@yrik I'm using pytest-specific features for this but basically I'm monkeypatching StrictRedis's `execute_command` with a function that calls `httpretty.disable()` first, then runs the real execute_command, then calls `httpretty.enable()`.

**Comment by gabrielfalcao** [comment]
This might have been fixed by @frankamp on #216
